diff --git a/configure.ac-orig b/configure.ac
index 66d1948..d2c283b 100755
--- a/configure.ac-orig
+++ b/configure.ac
@@ -16,7 +16,7 @@ AC_DEFUN([FIND_PARSER_SONAME],
 [
     AC_MSG_CHECKING([for lib_$1_parser_$2_elinux.3.0.so soname])
     path=`$PKG_CONFIG --variable=libdir libfslparser`
-    SONAME=$( $CC -print-file-name=$PKG_CONFIG_SYSROOT_DIR$path/lib_$1_parser_$2_elinux.so | \
+    SONAME=$( $CC -print-file-name=${STAGING_DIR_HOST}$path/lib_$1_parser_$2_elinux.so | \
 	    while read output; do objdump -p $output | \
 	    grep "SONAME" | \
 	    sed -e 's/ \+SONAME \+//'; done 2> /dev/null )
@@ -35,7 +35,7 @@ AC_DEFUN([FIND_ACODEC_SONAME],
 [
     AC_MSG_CHECKING([for lib_$1_wrap_$2_elinux.so soname])
     path=`$PKG_CONFIG --variable=libdir libfslaudiocodec`
-    SONAME=$( $CC -print-file-name=$PKG_CONFIG_SYSROOT_DIR$path/wrap/lib_$1_wrap_$2_elinux.so | \
+    SONAME=$( $CC -print-file-name=${STAGING_DIR_HOST}$path/wrap/lib_$1_wrap_$2_elinux.so | \
 	    while read output; do objdump -p $output | \
 	    grep "SONAME" | \
 	    sed -e 's/ \+SONAME \+//'; done 2> /dev/null )
